python/docs/source/usage.rst

```diff
@@ -399,7 +399,23 @@ only list the files to be deleted. Pass ``dry_run=False`` to actually delete fil
 Optimizing tables
 ~~~~~~~~~~~~~~~~~
 
-Optimizing tables is not currently supported.
+Optimizing a table will perform bin-packing on a Delta Table which merges small files
+into a large file. Bin-packing reduces the number of API calls required for read operations.
+Optimizing will increments the table's version and creates remove actions for optimized files.
+Optimize does not delete files from storage. To delete files that were removed, call :meth:`DeltaTable.vacuum`.
+
+Use :meth:`DeltaTable.optimize` to perform the optimize operation. Note that currently optimize only supports
+append-only workflows. Use with other workflows may corrupt your table state.
```

Comment:
I don't think this is true anymore ever since #632. We should just say that it will fail if there are concurrent transactions other than appends.

```suggestion
Use :meth:`DeltaTable.optimize` to perform the optimize operation. Note that this method will fail if a concurrent writer performs an operation that removes any files (such as an overwrite).
```